src/components/TextField.tsx

```diff
@@ -32,3 +36,86 @@ export const TextField = createRestyleComponent<Props, Theme>(
   restyleFunctions,
   TextInput,
 )
+
+type LabeledTextField = {
+  label: string
+  value: string
+  onChange?: (string) => void
+  onTap?: () => void
+  disabled?: boolean
+}
+
+export const LabeledTextField: React.FC<LabeledTextField> = ({
+  label,
+  value,
+  onChange,
+  onTap,
+  disabled,
+  ...props
+}) => (
+  <Box width="100%" py={4}>
+    <Text fontWeight="bold" color="darkGrey" pb={2}>
+      {label}
+    </Text>
+    <TextField
+      variant="login"
+      autoCapitalize="none"
+      autoCorrect={false}
+      onChangeText={(text) => onChange?.(text)}
+      onTouchEnd={() => onTap?.()}
+      value={value}
+      editable={disabled !== true}
+      {...props}
+    />
+  </Box>
+)
+
+export const LabeledDateField = ({ label, value, onChange, ...props }) => (
+  <Box width="100%" py={4}>
+    <Text fontWeight="bold" color="darkGrey" pb={2}>
+      {label}
+    </Text>
+    <Box
+      borderColor="purple"
+      borderWidth={2}
+      borderRadius="m"
+      alignItems="center"
+    >
+      <DatePicker mode="date" date={value} onDateChange={onChange} />
+    </Box>
+  </Box>
+)
+
+export const LabeledPickerField = ({
+  label,
+  value,
+  options = [],
+  onChange,
+  placeholder,
+}) => (
+  <Box width="100%" py={4}>
+    <Text fontWeight="bold" color="darkGrey" pb={2}>
+      {label}
+    </Text>
+    <Box
+      borderColor="purple"
+      borderWidth={2}
+      borderRadius="m"
+      alignItems="center"
+      justifyContent="center"
+    >
+      <RNPickerSelect
+        value={value}
+        fixAndroidTouchableBug
+        placeholder={placeholder}
+        textInputProps={{
+          height: 50,
+          borderRadius: 8,
+          paddingLeft: 16,
+        }}
+        onValueChange={onChange}
+        items={options}
+      />
+    </Box>
+  </Box>
+)
```

Comment:
Some of the components in this file feel like they don't belong inside `TextField`. Maybe generalize this file and call it `FormField` or something.

Reply:
Good Idea, Switched.